### PR TITLE
Update MySQLCluster

### PR DESCRIPTION
**Pull request by m-chrzan**
#41

### CODE REVIEW

File: operator/cluster/fake.go [hunk 1]
```diff
@@ -22,3 +22,8 @@ func (c *FakeClusterOperator) AddCluster(cluster *crv1.MySQLCluster) error {
 func (c *FakeClusterOperator) SetError(err error) {
 	c.err = err
 }
+
+func (c *FakeClusterOperator) UpdateCluster(newCluster *crv1.MySQLCluster) error {
+	// Just pretend we're updating a cluster. Do nothing.
+	return nil
```
Comment (earlgreyz) [inline operator/cluster/fake.go]: Maybe return `c.err` for testing

File: controller/cluster/controller.go [hunk 2]
```diff
@@ -65,6 +65,12 @@ func (c *clusterController) onUpdate(oldObj, newObj interface{}) {
 
 	logClusterEventBegin(newCluster, ClusterUpdated)
 
+	err := c.clusterOperator.UpdateCluster(newCluster)
+	if err != nil {
+		logging.LogCluster(newCluster).WithField("event", ClusterUpdated).Error(err)
+		return
```
Comment (earlgreyz) [inline controller/cluster/controller.go]: atm hooks should always run so don't return here and wrap logSuccess in an `else` statement

File: operator/cluster/operator.go [hunk 3]
```diff
@@ -73,8 +77,37 @@ func (c *clusterOperator) AddCluster(cluster *crv1.MySQLCluster) error {
...
+			newCluster,
+			"Failed update",
+			"The provided patch resulted in a StatefulSet update failure",
+		)
+		return errors.NewAggregate([]error{err, setStateErr})
+	}
+
+	updateErr := c.setClusterState(newCluster, "Successful update", "")
+
+	return updateErr
```
Comment (earlgreyz) [inline operator/cluster/operator.go]: Simply `return c.setClusterState(newCluster, "Successful update", "")`

File: operator/cluster/operator.go [hunk 4]
```diff
@@ -73,8 +77,37 @@ func (c *clusterOperator) AddCluster(cluster *crv1.MySQLCluster) error {
...
+}
+
+func (c *clusterOperator) setClusterState(cluster *crv1.MySQLCluster, state, message string) error {
+	cluster.Status.State = state
+	cluster.Status.Message = message
+	_, updateErr := c.clientset.CrV1().
+		MySQLClusters(cluster.ObjectMeta.Namespace).
+		Update(cluster)
```
Comment (earlgreyz) [inline operator/cluster/operator.go]:
```go
_, updateErr := c.clientset.CrV1().
 	MySQLClusters(cluster.ObjectMeta.Namespace).Update(cluster)
```
Is readable enough imho

File: operator/cluster/fake.go (hunk 1 above)
Comment (m-chrzan) [reply operator/cluster/fake.go]: Done.

File: controller/cluster/controller.go (hunk 2 above)
Comment (m-chrzan) [reply controller/cluster/controller.go]: Done.

File: operator/cluster/operator.go (hunk 3 above)
Comment (m-chrzan) [reply operator/cluster/operator.go]: Done.

File: operator/cluster/operator.go (hunk 4 above)
Comment (m-chrzan) [reply operator/cluster/operator.go]: Done.

**Pull request merged**